File: 2025-08/2025-08-13/fibonacci_sequence.py
def fibonacci_sequence(start_sequence, length):
    fibonacci = start_sequence[:]

    if len(fibonacci) >= length:
        return fibonacci[:length]

    fibonacci.append(fibonacci[-2] + fibonacci[-1])

    return fibonacci_sequence(fibonacci, length)

print(fibonacci_sequence([0, 1], 20))
print(fibonacci_sequence([21, 32], 1))
print(fibonacci_sequence([0, 1], 0))
print(fibonacci_sequence([10, 20], 2))
print(fibonacci_sequence([123456789, 987654321], 5))

File: 2025-08/2025-08-13/fibonacciSequence.js
function fibonacciSequence(startSequence, length) {
  const fibonacci = [...startSequence];

  if (fibonacci.length >= length)
    return fibonacci.slice(0, length);

  fibonacci.push(fibonacci[fibonacci.length - 2] + fibonacci[fibonacci.length - 1]);

  return fibonacciSequence(fibonacci, length);
}

console.log(fibonacciSequence([0, 1], 20));
console.log(fibonacciSequence([21, 32], 1));
console.log(fibonacciSequence([0, 1], 0));
console.log(fibonacciSequence([10, 20], 2));
console.log(fibonacciSequence([123456789, 987654321], 5));